src/main/java/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,201 @@
+package internet.shop.dao.jdbc;
+
+import internet.shop.dao.ShoppingCartDao;
+import internet.shop.exceptions.DataProcessingException;
+import internet.shop.lib.Dao;
+import internet.shop.model.Product;
+import internet.shop.model.ShoppingCart;
+import internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        ShoppingCart cart = null;
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ? AND is_deleted = FALSE";
+        try (Connection conn = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = conn.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cart = getCart(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get cart by user id = "
+                    + userId, e);
+        }
+        return Optional.ofNullable(cart);
+    }
+
+    @Override
+    public double getTotalPrice(Long id) {
+        double total = 0;
+        String query = "SELECT SUM(price) as total FROM products join shopping_carts_products scp "
+                + "on products.product_id = scp.product_id where cart_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                total = resultSet.getDouble("total");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get all carts!", e);
+        }
+        return total;
+    }
```

Comment:
we can make this logic out of base. In service layer instead of dao

Reply:
I placed this logic here because it responsible for calculating the shopping cart parameter but not for changing the shopping cart